### PR TITLE
Made webserver a separate class

### PR DESCRIPTION
**Pull request by schung507**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,11 +2,11 @@ CXX=g++
 CXXOPTIMIZE= -O2
 CXXFLAGS= -g -Wall -pthread -std=c++0x $(CXXOPTIMIZE)
 GTEST_DIR=googletest/googletest
-SERVERCLASSES=config_parser.cc
+SERVERCLASSES=config_parser.cc Webserver.h webserver_main.cc
```
Comment (llam15) [inline Makefile]: Capitalize the w in webserver_main.cc

File: Webserver.cc [hunk 2]
```diff
@@ -1,88 +1,71 @@
...
+	    // Get the port number.
+	    port = std::stoi(config_out.statements_[0]->tokens_[1]);
+	} else {
+	    std::cerr << "Error: Could not parse config file.\n";
+	}
+
+	return result;
+}
+
+
```
Comment (llam15) [inline Webserver.cc:29]: Please remove extraneous new lines

File: Webserver.cc [hunk 3]
```diff
@@ -1,88 +1,71 @@
...
-}
\ No newline at end of file
+#include "Webserver.h"
+
+#include <cstdlib>
+#include <utility>
+#include <iostream>
+#include <thread>
+#include <boost/asio.hpp>
+
```
Comment (llam15) [inline Webserver.cc:10]: Let's alphabetize this

File: Webserver.cc [hunk 4]
```diff
@@ -1,88 +1,71 @@
...
+const int max_length = 4096;
+
+using boost::asio::ip::tcp;
+
+bool Webserver::parse_config(const char* file_name){
+
```
Comment (llam15) [inline Webserver.cc:16]: Please remove extraneous new line

File: Webserver.h [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+#include <boost/asio.hpp>
+
+class Webserver {
+
+public:
+	bool parse_config(const char* file_name);
+	void run_server(boost::asio::io_service& io_service);
+	void session(boost::asio::ip::tcp::socket sock); 
+
+
```
Comment (llam15) [inline Webserver.h]: Please remove extraneous new lines

File: Webserver_main.cc [hunk 6]
```diff
@@ -0,0 +1,32 @@
+// Based on Boost library blocking_tcp_echo_server example.
+// http://www.boost.org/doc/libs/1_55_0/doc/html/boost_asio/example/cpp11/echo/blocking_tcp_echo_server.cpp
```
Comment (llam15) [inline Webserver_main.cc]: Let's move these comments into Webserver.cc

File: Webserver_main.cc [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+#include <cstdlib>
+#include <iostream>
+#include <thread>
+#include <utility>
+#include <boost/asio.hpp>
+#include "Webserver.h"
+
+int main(int argc, char* argv[]) {
+
```
Comment (llam15) [inline Webserver_main.cc:9]: Please remove extraneous new line

**Review by llam15 — CHANGES_REQUESTED** [review]

**Review by tyhchoi — APPROVED** [review]
yea looks good

**Pull request merged**